File: job-offer/src/main/java/com/oxygensend/joboffer/application/job_offer/dto/view/AddressView.java
package com.oxygensend.joboffer.application.job_offer.dto.view;


import com.fasterxml.jackson.annotation.JsonInclude;
import com.oxygensend.joboffer.domain.entity.Address;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record AddressView(String postCode, String city, Double lon, Double lat) {

    public static AddressView from(Address address) {
        return new AddressView(address.postCode(), address.city(), null, null);
    }

    public static AddressView withCoords(Address address) {
        return new AddressView(address.postCode(), address.city(), address.lon(), address.lat());
    }
}

File: static-data/src/main/kotlin/com/oxygensend/staticdata/application/address/AddressView.kt
package com.oxygensend.staticdata.application.address

import com.oxygensend.staticdata.domain.Address

data class AddressView(val city: String, val lon: Double?, val lat: Double?) {
    companion object {
        fun from(address: Address) = AddressView(address.city, address.lon, address.lat)
    }
}